### PR TITLE
Stop sending packets if F-family error encountered

### LINKED ISSUE #430: STREAM Sender: Unreachable destinations will retry until timeout but should short-circuit
If a STREAM sender tries to send a payment on a link but the Connector can't forward the packets (i.e., the StreamSender gets an `F02 Unreachable` error, the Sender should stop sending STREAM packets. However, the current implementation retries every 10ms until the overall sendMoney timeout is hit.

### PR DESCRIPTION
**Pull request by nkramer44**
Fixes #430

### CODE REVIEW

File: stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java [hunk 1]
```diff
@@ -482,17 +482,26 @@ default int totalPackets() {
...
+                                                                     StreamPacket streamPacket) {
       InterledgerResponsePacket response = link.sendPacket(preparePacket);
-      response.handle((fulfill) -> {}, (reject) -> {
-        if (reject.getCode().equals(F00_BAD_REQUEST)) {
-          unrecoverableErrorEncountered.set(true);
-        }
+      response.handle(
+        (fulfill) -> {},
+        (reject) -> {
+          if (!reject.getCode().equals(F08_AMOUNT_TOO_LARGE) &&
```
Comment (kincaidoneil) [inline stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java]: Shouldn't this also keep retrying on F99 errors?

Comment (nkramer44) [reply stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java]: @sappenin would know better...  

The Java receiver only returns an F99 if the prepare amount < stream amount, but the Java sender doesn't have any retry behavior to deal with F99s.  Maybe it should?

Comment (kincaidoneil) [reply stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java]: There are other cases where the JavaScript STREAM receiver returns F99 errors, too, which is something to consider.

> The Java receiver only returns an F99 if the prepare amount < stream amount

But should the payment fail if only a single packet doesn't meet the minimum exchange rate? Or should there be multiple attempts?

Rust and JS retry like normal on F99s, but I think we probably need more sophisticated behavior for the exchange rate issues.

Comment (nkramer44) [reply stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java]: Agreed. For now, our main motivation for short circuiting is to reduce our log output/error alerting, which is mainly happening on F02s. I will narrow this condition to only short circuit on F02s, and we can discuss what we want to do on F99s.

**Comment by sappenin** [comment]
This PR has been superseded by https://github.com/hyperledger/quilt/pull/443

**Pull request closed without merging**